src/huggingface_hub/hf_api.py

```diff
@@ -9940,6 +9944,488 @@ def auth_check(
+        token: Union[bool, str, None] = None,
+        _repo: Optional[str] = None,
+    ) -> JobUrl:
+        """
+        Run a UV script Job on Hugging Face infrastructure.
+
+        Args:
+            script (`str`):
+                Path or URL of the UV script.
+
+            script_args (`List[str]`, *optional*)
+                Arguments to pass to the script.
+
+            dependencies (`List[str]`, *optional*)
+                Dependencies to use to run the UV script.
+
+            python (`str`, *optional*)
+                Use a specific Python version. Default is 3.12.
+
+            env (`Dict[str, Any]`, *optional*):
+                Defines the environment variables for the Job.
+
+            secrets (`Dict[str, Any]`, *optional*):
+                Defines the secret environment variables for the Job.
+
+            flavor (`str`, defaults to `"cpu-basic"`):
+                "Flavor for the hardware, as in Hugging Face Spaces.
+
+            timeout (`Union[int, float, str]`, *optional*):
+                Max duration for the Job: int/float with s (seconds, default), m (minutes), h (hours) or d (days).
+                Example: `300` or `"5m"` for 5 minutes.
+
+            token `(Union[bool, str, None]`, *optional*):
+                A valid user access token. If not provided, the locally saved token will be used, which is the
+                recommended authentication method. Set to `False` to disable authentication.
+                Refer to: https://huggingface.co/docs/huggingface_hub/quick-start#authentication.
+
+        Example:
+
+            ```python
+            >>> from huggingface_hub import run_uv_job
+            >>> script = "https://raw.githubusercontent.com/huggingface/trl/refs/heads/main/trl/scripts/sft.py"
+            >>> run_uv_job(script, dependencies=["trl"], flavor="a10g-small")
+            ```
+        """
+
+        if script.startswith("http://") or script.startswith("https://"):
+            # Direct URL execution - no upload needed
+            script_url = script
+        else:
+            # Local file - upload to HF
+            script_path = Path(script)
+
+            def _determine_repository():
+                """Determine which repository to use for the script."""
+                # Use provided repo
+                if _repo:
+                    repo_id = _repo
+                    if "/" not in repo_id:
+                        username = self.whoami(token=token)["name"]
+                        repo_id = f"{username}/{repo_id}"
+                    return repo_id
+
+                # Create ephemeral repo
+                username = self.whoami(token=token)["name"]
+                timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
+
+                # Simple hash for uniqueness
+                script_hash = hashlib.md5(Path(script).read_bytes()).hexdigest()[:8]
+
+                return f"{username}/huggingface-cli-jobs-uv-run-{timestamp}-{script_hash}"
```

Comment:
```suggestion
                return f"{username}/hf-jobs-uv-run-{timestamp}-{script_hash}"
```

(let's make it shorter)